src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/custom.py

```diff
@@ -1452,8 +1452,20 @@ def _download_secret(vault_base_url, secret_name, file_path, encoding=None, secr
                         crypto.FILETYPE_PEM, cert))
             f_pem.close()
         except Exception as ex:  # pylint: disable=broad-except
-            if os.path.isfile(file_path):
-                os.remove(file_path)
+            if os.path.isfile(pem_path):
+                os.remove(pem_path)
+            raise ex
+
+    if pfx_path is not None:
+        try:
+            import base64
+            decoded = base64.b64decode(secret_value)
+            p12 = crypto.load_pkcs12(decoded)
+            with open(pfx_path, 'wb') as f:
+                f.write(decoded)
+        except Exception as ex:  # pylint: disable=broad-except
```

Comment:
Why do you disable the pylint rule here? Can you narrow the exception types are expected here?

Reply:
this logic is from Key vault. any exception happens we don't want output the cert file. 

https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py#L554

Reply:
A similar code snippet does exist in current code base. But it doesn't necessarily make it right. We're going to revisit all the `broad-except` pylint rule disable in the future.

Reply:
Filed issue: https://github.com/Azure/azure-cli/issues/4486